datarepo-clienttests/src/main/java/runner/config/TestConfiguration.java

```diff
@@ -71,45 +70,19 @@ public static TestConfiguration fromJSONFile(String resourceFileName) throws Exc
    * of the objects, for example by parsing the string values in the JSON object.
    */
   public void validate() {
+    LOG.debug("Validating the server, Kubernetes and application specifications");
     server.validate();
     kubernetes.validate();
     application.validate();
 
+    LOG.debug("Validating the test script specifications");
     for (TestScriptSpecification testScript : testScripts) {
       testScript.validate();
     }
 
+    LOG.debug("Validating the test user specifications");
     for (TestUserSpecification testUser : testUsers) {
       testUser.validate();
     }
   }
-
-  public void display() {
-    System.out.println("Test configuration: " + name);
-    System.out.println("  Description: " + description);
-
-    System.out.println();
-    server.display();
-
-    System.out.println();
-    kubernetes.display();
-
-    System.out.println();
-    application.display();
-
-    System.out.println();
-    System.out.println("  Billing account: " + billingAccount);
-
-    for (TestScriptSpecification testScript : testScripts) {
-      System.out.println();
-      testScript.display();
-    }
-
-    for (TestUserSpecification testUser : testUsers) {
-      System.out.println();
-      testUser.display();
-    }
-
-    System.out.println();
-  }
```

Comment:
What is the goal behind moving away from this pattern [a display() method for each specification class]?

I have mixed feeling about this transition. I think the new SpecificationInterface's display() function is helpful to be able to spit out whole json objects for debugging.  But, I like that this existing pattern allows specific display() methods pick out important info and display it in a readable format.

Reply:
I swapped out the text format for JSON format mainly because when I changed to use the logger instead of stdout.println, all the nice indenting got messed up because the logger prepends the class name. For example, it went **from this**:
Test configuration: BasicUnauthenticated
  Description: Two users check the service status repeatedly, concurrently. No authentication required.

Server: mmdev
  description: Developer namespace (mm) in Jade dev cluster. Supports deploying the server with a modular Helm chart and modifying Kubernetes post-deployment.
  uri: https://jade-mm.datarepo-dev.broadinstitute.org
  clusterName: gke_broad-jade-dev_us-central1_dev-master
  clusterShortName: dev-master
  region: us-central1
  project: broad-jade-dev
  namespace: mm
  skipKubernetes: false
  skipDeployment: false

Deployment Script: ModularHelmChart
  name: ModularHelmChart
  parameters: file:///Users/marikomedlock/Workspaces/datarepo-helm-definitions/dev/mm/datarepo-api.yaml

Kubernetes: 
  numberOfInitialPods: 1

Application: 
  maxStairwayThreads: 20
  maxBulkFileLoad: 1000000
  loadConcurrentFiles: 4
  loadConcurrentIngests: 2
  loadHistoryCopyChunkSize: 1000
  loadHistoryWaitSeconds: 2

  Billing account: 00708C-45D19D-27AAFA

**to this:**
10:47:07.989 [main] INFO runner.config.TestConfiguration - Test configuration: BasicUnauthenticated
10:47:07.990 [main] INFO runner.config.TestConfiguration -   Description: Two users check the service status repeatedly, concurrently. No authentication required.
10:47:07.990 [main] INFO runner.config.ServerSpecification - Server: mmdev
10:47:07.990 [main] INFO runner.config.ServerSpecification -   description: Developer namespace (mm) in Jade dev cluster. Supports deploying the server with a modular Helm chart and modifying Kubernetes post-deployment.
10:47:07.990 [main] INFO runner.config.ServerSpecification -   uri: https://jade-mm.datarepo-dev.broadinstitute.org
10:47:07.990 [main] INFO runner.config.ServerSpecification -   clusterName: gke_broad-jade-dev_us-central1_dev-master
10:47:07.990 [main] INFO runner.config.ServerSpecification -   clusterShortName: dev-master
10:47:07.990 [main] INFO runner.config.ServerSpecification -   region: us-central1
10:47:07.990 [main] INFO runner.config.ServerSpecification -   project: broad-jade-dev
10:47:07.990 [main] INFO runner.config.ServerSpecification -   namespace: mm
10:47:07.990 [main] INFO runner.config.ServerSpecification -   skipKubernetes: false
10:47:07.990 [main] INFO runner.config.ServerSpecification -   skipDeployment: true
10:47:07.990 [main] INFO runner.config.KubernetesSpecification - Kubernetes: 
10:47:07.990 [main] INFO runner.config.KubernetesSpecification -   numberOfInitialPods: 1
10:47:07.990 [main] INFO runner.config.ApplicationSpecification - Application: 
10:47:07.990 [main] INFO runner.config.ApplicationSpecification -   maxStairwayThreads: 20
10:47:07.990 [main] INFO runner.config.ApplicationSpecification -   maxBulkFileLoad: 1000000
10:47:07.990 [main] INFO runner.config.ApplicationSpecification -   loadConcurrentFiles: 4
10:47:07.990 [main] INFO runner.config.ApplicationSpecification -   loadConcurrentIngests: 2
10:47:07.990 [main] INFO runner.config.ApplicationSpecification -   loadHistoryCopyChunkSize: 1000
10:47:07.990 [main] INFO runner.config.ApplicationSpecification -   loadHistoryWaitSeconds: 2
10:47:07.990 [main] INFO runner.config.TestConfiguration -   Billing account: 00708C-45D19D-27AAFA

It's also a little easier to maintain because you don't have to remember to update the display method whenever you add a new property, but the main reason is because the logger formatting made the text pretty unwieldy, in my opinion. I'd be open to supporting two different display modes text vs json, where we print to standard out vs use the logger, if you think the previous format was more useful. Let me know what you think.

Reply:
Oh, ok - that makes sense, thanks for explaining!  And no, I think it's better to stick with one logging method.